telegram-bot/main.go

```diff
@@ -28,6 +30,29 @@ func ButtonLinks(ID int64, ButtonText string, ButtonUrl string, MessageText stri
 	msg.ReplyMarkup = button
 	bot.Send(msg)
 }
+//Streams tweets with the help of Twitter API , bot sends the link of any new tweet or tweet related activity to the specified group on telegram.
+func tweet() {
+	//Initializing twitter API
+	config := oauth1.NewConfig(os.Getenv("TWITTER_CONSUMER_KEY"), os.Getenv("TWITTER_CONSUMER_SECRET"))
+	token := oauth1.NewToken(os.Getenv("TWITTER_ACCESS_TOKEN"), os.Getenv("TWITTER_ACCESS_SECRET"))
+	httpClient := config.Client(oauth1.NoContext, token)
+	client := twitter.NewClient(httpClient)
+
+	params := &twitter.StreamFilterParams{
+		Follow: []string{os.Getenv("OSDC_TWITTER_ID")},
+	}
+	stream, _ := client.Streams.Filter(params)
+
+	demux := twitter.NewSwitchDemux()
+	demux.Tweet = func(tweet *twitter.Tweet) {
+		fmt.Println(tweet.IDStr)
+		Group_ID, _ := strconv.ParseInt(os.Getenv("OSDC_GROUP_ID"), 10, 64)
+
+		bot.Send(tbot.NewMessage(Group_ID, "New Tweet Alert: \n https://twitter.com/osdcjiit/status/"+tweet.IDStr))
```

Comment:
@PrimalPimmy Can you parse the tweet link into a button? 

The function https://github.com/osdc/bots/blob/974680f6dd81f3fd284428cd5e3a8eb4757271a4/telegram-bot/main.go#L21

Reference implementation https://github.com/osdc/bots/blob/974680f6dd81f3fd284428cd5e3a8eb4757271a4/telegram-bot/paste.go#L35

Reply:
If we put Button for the tweet alert, wouldn't it hide the contents of the tweet? We need the contents of the tweet to show up in the chat too.

Reply:
Makes sense.